pivottable/index.js

```diff
@@ -0,0 +1,57 @@
+window.onerror = (err) => {
+  console.trace();
+  alert(String(err));
+};
+
+grist.ready({
+  requiredAccess: 'read table'
+});
+
+const urlParams = new URLSearchParams(window.location.search);
+const rows = urlParams.get('rows')?.split(',');
+const cols = urlParams.get('cols')?.split(',');
+const vals = urlParams.get('vals')?.split(',');
+const aggregatorName = urlParams.get('aggregatorName');
+const rendererName = urlParams.get('rendererName');
+
+function wavg (n) {
+  if (!n) { return; }
+  n = n.filter(([note]) => typeof (note) === 'number');
+  if (n.length) { return n.map(([note, coef]) => note * coef).reduce((a, b) => a + b) / n.map(([_note, coef]) => coef).reduce((a, b) => a + b); }
+}
+
+function weightedAverage ([val, coef]) {
+  return (_data, _rowKey, _colKey) => ({
+    values: [],
+    push: function (rec) { this.values.push([rec[val], rec[coef]]); },
+    value: function () { return wavg(this.values); },
+    format: function (x) { return (Math.round(x * 100) / 100).toFixed(2); },
+    numInputs: 2
+  });
+}
+
+const aggregators = {
+  'Weighted Average': weightedAverage
+};
+
+grist.onRecords(rec => {
+  console.log(rec);
```

Comment:
You can use Grist API to save and restore this settings. For example:

```js
grist.onRecords(async rec => {
  const { rows, cols, vals, aggregatorName, rendererName } = await grist.getOption('settings') ?? {};
  let initialRender = true;
  $('#table').pivotUI(
    rec,
    {
      rows,
      cols,
      vals,
      onRefresh: function (config) {
        if (initialRender) {
          initialRender = false;
          return;
        }
        const { rows, cols, vals, aggregatorName, rendererName } = config;
        grist.setOption('settings', { rows, cols, vals, aggregatorName, rendererName });
      },
```

Reply:
Done. Should I keep the url parameters as an alternative, or delete them ?